### PR TITLE
Netlify deployment

### LINKED ISSUE #4117: Migrate deployment of our documentation site over to Netlify
**Short description**
As a Developer I want to migrate the deployment of our documentation site over to Netlify from GCP so that 
I can
- simplify the deployment process
- speed up deployment time and reduce costs
- help get rid of old version folders being served from the bucket even when they are deleted from our repository
- improve developer productivity since deploy previews over pull requests will no longer take several minutes

Acceptance criteria
- [ ] Add a new workflow for documentation deployment for Netlify
- [ ] Create and setup the documentation site over Netlify
- [ ] Setup relevant env vars, deploy keys, domains etc., for production deploys
- [ ] Make the switch seamless and avoid downtime

**Design materials**
- Screenshots
- Design diagrams
- WEP 
- PR/FAQ

**Work breakdown**
(optional)Task breakdown of the work
- [ ] _create X_
- [ ] _design X_
- [ ] _..._

### PR DESCRIPTION
**Pull request by Himangini**
Part of #4117 

**What changed?**
Deployment workflow to deploy documentation over Netlify when release happens.
Todo - Test over a production branch

**Why was this change made?**
See issue

**How was this change implemented?**

**How did you validate the change?**
Testing deployment manually with Netlify  + deploy previews

**Release notes**
Todo

**Documentation Changes**
Todo

### CODE REVIEW

**Review by enekofb — COMMENTED** [review]
Looked at the PR but I think I might miss some context to review: 

looked to the referenced issue where mentioned 

>Add a new workflow for documentation deployment for Netlify 

but could not understand the expected AC for the workflow. please let me know more about it. For example, is under scope the development or staging?

Also, could you add some info about how validation happened?

File: .github/workflows/docs-netlify.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    types: [published] # remove auto deploying docs everytime pull request is merged
+  workflow_dispatch:
+
+permissions:
+  contents: write 
+
+jobs:
+  publish-docs:
+    name: Publish docs to Netlify
+    if: ${{ !github.event.release.prerelease }}
```
Comment (makkes) [inline .github/workflows/docs-netlify.yaml:14]: Will this work when manually running the workflow?

File: .github/workflows/docs-netlify.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        env:
+          USE_SSH: true
+          GIT_USER: git
+          GA_KEY: ${{ secrets.GA_KEY }}
+          ALGOLIA_API_KEY: ${{ secrets.ALGOLIA_API_KEY }}
+          NODE_OPTIONS: "--max-old-space-size=4096"
+        run: | 
+          git config --global user.email "corporate-it+weaveworks-docs-bot@weave.works"
+          git config --global user.name "weaveworks-docs-bot"
+          git switch docs-production
```
Comment (makkes) [inline .github/workflows/docs-netlify.yaml:34]: I suppose the way you imagine hotfixes to the docs to land on the website is to issue PRs against `docs-production` which is ok in my opinion. That way we can use `main` as dev branch for the next release. But we need to cleary document this new process.

**Review by makkes — COMMENTED** [review]
I suppose the only way we can actually test this new workflow is by merging it, as with most other changes in GHA.

@Himangini can you set up deploy previews in Netlify so that we can already see the results here?

File: .github/workflows/docs-netlify.yaml (hunk 1 above)
Comment (Himangini) [reply .github/workflows/docs-netlify.yaml:14]: yeah, it should work.

**Comment by Himangini** [comment]
> I suppose the only way we can actually test this new workflow is by merging it, as with most other changes in GHA.
> 
> @Himangini can you set up deploy previews in Netlify so that we can already see the results here?

@makkes the deploy preview is set up.

**Comment by casibbald** [comment]
No commercial netlify license. closing

**Pull request closed without merging**